Mathlib/Data/Sym/Basic.lean

```diff
@@ -532,6 +532,45 @@ theorem mem_append_iff {s' : Sym α m} : a ∈ s.append s' ↔ a ∈ s ∨ a ∈
   Multiset.mem_add
 #align sym.mem_append_iff Sym.mem_append_iff
 
+/-- An element of type α can be seen as an element of Sym α 1 --/
+def one {α : Type*} (i : α) : Sym α 1 := ⟨[i], by simp only [Multiset.coe_singleton,
+  Multiset.card_singleton]⟩
+
+lemma one_injective : Function.Injective (Sym.one : α → Sym α 1) := by
+  intros i j h
+  have hi : (Sym.one i).1 = [i] := by
+      simp only [Sym.one, Multiset.coe_singleton]
+  have hj : (Sym.one j).1 = [j] := by
+    simp only [Sym.one, Multiset.coe_singleton]
+  rw [h] at hi
+  rw [hi] at hj
+  apply List.singleton_inj.mp
+  exact Multiset.coe_eq_singleton.mp hj
+
+lemma singleton_of_len_one (s : Sym α 1) : ∃! i, Sym.one i = s := by
+  have : ∃ i, s.toMultiset = {i} := by
+    simp_rw [Sym] at s
+    apply Multiset.card_eq_one.mp
+    exact s.2
+  obtain ⟨i, hi⟩ := this
+  have : s = one i := by
+    simp only [Sym.one, Multiset.coe_singleton]
+    simp_rw [← hi]
+    rfl
+  use i
+  constructor
+  · exact this.symm
+  · intro j hj
+    apply Sym.one_injective
+    rw [hj]
+    exact this
+
+lemma one_bijective : Function.Bijective (Sym.one : α → Sym α 1) := by
+  rw [Function.bijective_iff_existsUnique]
+  exact Sym.singleton_of_len_one
```

Comment:
```suggestion
variable (α) in
lemma one_bijective : Function.Bijective (one : α → Sym α 1) := by
  refine ⟨one_injective _, fun x ↦ ?_⟩
  obtain ⟨i, hi⟩ := Multiset.card_eq_one.1 x.2
  exact ⟨i, ext <| by simp [one_coe, ← hi]⟩
```
`singleton_of_len_one` is just a strange way of saying that the function is bijective, so I removed it. Also here I went full golf.

You can go one step further if you want, and prove directly
```lean
variable (α) in
lemma one_bijective : Function.Bijective (one : α → Sym α 1) := by
  refine ⟨fun _ _ h ↦ Multiset.singleton_inj.1 <| by simp [← one_coe, h], fun x ↦ ?_⟩
  obtain ⟨i, hi⟩ := Multiset.card_eq_one.1 x.2
  exact ⟨i, ext <| by simp [one_coe, ← hi]⟩
```